Eobs 214 news process changes no score and low clinical risk

homostellaris opened the pull request:
Mainly extracting methods and refactoring in this layer to make things reusable in nh_configurations.

Changes to NEWS policy for no risk and low risk observations.
# If no risk obs
- When patient was admitted less than 7 days ago, set frequency to 12 hours.
- When patient was admitted more than 7 days ago, set frequency to 24 hours.
- When patient has been admitted more than 7 days ago, create a 'Select Frequency' task assigned to the nurse.
- When 'weekly' frequency is set from 'Select Frequency' task, create a 'Weekly Frequency Agreed With Medical Staff' task.
# If low risk obs
- Create a 'Select Frequency' task assigned to the nurse.
- Provide different options for 'Select Frequency' task based on if patient was admitted less than 7 days ago.

Commit:
'Select Frequency' task is created after a low
risk observation. But it does not yet show different
frequencies when patient has been admitted over 7
days ago.

Changed patient_id to hospital_number as that is
more accurate and was confusing in test methods
when dealing with actual patient_id for the model
in the same method.

Commit:
Mainly extracting methods and refactoring in this layer
to make things reusable in nh_configurations.

Changes to NEWS policy for no risk and low risk observations.

IF NO RISK OBS
* When patient was admitted less than 7 days ago,
  set frequency to 12 hours.
* When patient was admitted more than 7 days ago,
  set frequency to 24 hours.
* When patient has been admitted more than 7 days ago,
  create a 'Select Frequency' task assigned to the nurse.
* When 'weekly' frequency is set from 'Select Frequency'
  task, create a
  'Weekly Frequency Agreed With Medical Staff' task.

IF LOW RISK OBS
* Create a 'Select Frequency' task assigned to the nurse.
* Provide different options for 'Select Frequency' task
  based on if patient was admitted less than 7 days ago.

Commit:
Merge branch 'develop' into EOBS-214-news-process-changes-no-score-and-low-clinical-risk

Gimpneek commented on nh_observations/nh_clinical_extension.py:
```diff
@@ -174,7 +174,9 @@ def trigger_notifications(self, cr, uid, values, context=None):
                     'patient_id': values.get('patient_id')
                 }
-                if n['model'] == 'frequency':
+                if n['model'] == 'frequency' \
```
Cleaner to use `if n['model'] in ['frequency', 'select_frequency', 'weekly_frequency_agreed']` ?

Gimpneek commented on nh_ews/ews.py:
```diff
@@ -781,18 +810,27 @@ def get_last_case(self, cr, uid, patient_id, context=None):
-            self._POLICY['ranges'], activity.data_ref.score)])
-        case = 2 if activity.data_ref.three_in_one and case < 3 else case
-        return case
+        if ews_ids:
```
Is this method no longer returning False if there's no ews_ids?

Gimpneek replied:
Need to return falsde if no ews_ids in get last obs and also return false in get last case

Gimpneek commented on nh_observations/frequencies.py:
```diff
@@ -0,0 +1,33 @@
+    ]
+
+def minutes_only():
+    return [minutes for minutes, _ in as_list()]
```
Do we need to reference both tuple values in loop? 
Maybe [item[0] for item in as_list()]?
Not suggesting we profile it but something to think about

Commit:
Fixed flake8 errors.

Commit:
Merge branch 'develop' into EOBS-214-news-process-changes-no-score-and-low-clinical-risk

Commit:
Changed frequencies to reflect agreed wording
and updated failing tests accordingly.

Changed a conditional with several `or`
operators to use an `in` operator with
an array to make it more concise.

Commit:
Response to code review.

Incorrect refactoring meant that `get_last_obs_activity()`
no longer returned `False` when no activity was found.
Although I don't agree with this return value it was
put back in to maintain backwards compatibility.

Commit:
Moved methods to place they can be more
easily reused.

Commit:
Fixed flake8 errors.

Gimpneek commented:
:shipit:

Commit:
Added method for reuse.

Gimpneek commented on nh_observations/notifications.py:
```diff
@@ -129,6 +129,11 @@ def complete(self, cr, uid, activity_id, context=None):
         }
     ]
 
+    def set_form_description_frequencies(self, available_frequencies):
```
How does this play with https://github.com/NeovaHealth/openeobs/blob/develop/nh_ews/notifications.py#L139 ?

Setting the observation frequencies on the nh_observation object will mean that the change to the form description in nh_ews will most likely override this dictionary anyways

Gimpneek replied:
I'm not entirely why nh_observation has the Review Frequency defined anyways as this is currently an EWS only notification

moderatemisbehaviour replied:
I think the method should still work okay even if `_form_description` is overridden lower down because it's being called from lower down with `self`. This method is only used in my code in nh_configurations currently but seemed best to put it here for reuse as it could be used anytime with a *nh.clinical.notification.frequency* object.

Commit:
Merge branch 'develop' into EOBS-214-news-process-changes-no-score-and-low-clinical-risk

Commit:
Merge branch 'develop' into EOBS-214-news-process-changes-no-score-and-low-clinical-risk

Nayira commented:
Test of feature branch passed - :shipit: !
